### DISCUSSION TITLE
Shall we consider adding more unary and binary operator functionalities?

### DISCUSSION THREAD

**Discussion by zakean** [dotnet/csharplang#8367] [Language Ideas] [answered]
I hope for the addition of more unary and binary operator functionalities, allowing for more concise and expressive syntax in C#. 

For example, the current `if` statement usage is:

```csharp
if (bCondition)
{
    someVar = "Hello";
}
```

This can be slightly simplified:

```csharp
if (bCondition)
    someVar = "Hello";
```

However, I propose supporting a syntax like the following:

```csharp
bool bCondition = true;
object obj = null;

bCondition ?? someVar = "Hello";
obj ?? bCondition = false;
(!bCondition) ?? someVar = "No";
```

This would enable more streamlined and readable code, reducing the need for verbose conditional statements.

**Comment by HaloFour** [comment, marked as answer]
At best it seems that you're only saving two characters over the current syntax of:

```cs
// proposed
bCondition ?? someVar = "Hello";

// already legal
if (bCondition) someVar = "Hello";
```

**Comment by zms9110750** [comment]
I support adding more optional symbols as operators. Anyway, they are essentially just shorthand for calling methods.
If the functionality of this symbol has already been restricted, then it needs to be considered.
Your proposal doesn't look very good.
